azure-ipam/windows.Dockerfile

```diff
@@ -0,0 +1,32 @@
+ARG ARCH
+ARG DROPGZ_VERSION=v0.0.12
+ARG OS
+ARG OS_VERSION
+
+FROM --platform=linux/${ARCH} mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS azure-ipam
+ARG OS
+ARG VERSION
+WORKDIR /azure-ipam
+COPY ./azure-ipam .
+RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-ipam -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" .
+
+FROM --platform=linux/${ARCH} mcr.microsoft.com/cbl-mariner/base/core:2.0 AS compressor
+ARG OS
+WORKDIR /payload
+COPY --from=azure-ipam /go/bin/* /payload
+COPY --from=azure-ipam /azure-ipam/*.conflist /payload
+RUN cd /payload && sha256sum * > sum.txt
+RUN gzip --verbose --best --recursive /payload && for f in /payload/*.gz; do mv -- "$f" "${f%%.gz}"; done
+
+FROM --platform=linux/${ARCH} mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS dropgz
```

Comment:
Same here

Reply:
Oh, maybe not, I see it is like this in dropgz windows today, as well as cniTest
If that's the case then I can just approve, lgtm

Reply:
correct, we build _on_ Linux _for_ Windows, because Go has good cross-compilation and building _on_ Windows containers in CI is super unreliable.